de.dlr.sc.virsat.uiengine.ui/src/de/dlr/sc/virsat/uiengine/ui/editor/snippets/general/UiSnippetIDescription.java

```diff
@@ -0,0 +1,137 @@
+ *
+ */
+public class UiSnippetIDescription extends AUiEStructuralFeatureSectionSnippet implements IUiSnippet {
+
+	public static final String SECTION_HEADING = "Description Section";
+	public static final String DESCRIPTION_FIELD = "Description";
+	
+	private static final int UI_LAYOUT_NR_COLUMNS =  2;	
+	
+	private EsfMarkerImageProvider emip;
+	
+	private Text textDescription;
+	private Label labelPropertyIcon;
+
+	
+	/**
+	 * Constructor of the snippet initializing internal helpers etc.
+	 */
+	public UiSnippetIDescription() {
+		super(GeneralPackage.Literals.IDESCRIPTION__DESCRIPTION);
+		emip = new EsfMarkerImageProvider();
+	}
+	
+	@Override
+	public void createSwt(FormToolkit toolkit, EditingDomain editingDomain, Composite composite, EObject initModel) {
+		super.createSwt(toolkit, editingDomain, composite, initModel);
+		
+		Composite sectionBody = createSectionBody(toolkit, SECTION_HEADING, null, UI_LAYOUT_NR_COLUMNS);
+		
+		// Divide left column into new parts
+		Composite labelWithIcon = toolkit.createComposite(sectionBody);
+		labelWithIcon.setLayout(new GridLayout(2, false));
+		
+		labelPropertyIcon = toolkit.createLabel(labelWithIcon, "");
+		Label labelDescription = toolkit.createLabel(labelWithIcon, DESCRIPTION_FIELD);
+		textDescription = toolkit.createText(sectionBody, "", SWT.MULTI | SWT.V_SCROLL | SWT.BORDER | SWT.WRAP);
+		
+		setUpIcon();
+		setUpLabel(labelDescription);		
+		setUpText(editingDomain);
+	    
+		checkWriteAccess(textDescription);
+	}
+
+	/**
+	 * Method to set up the Text with specific Grid
+	 * @param editingDomain the editingDomain
+	 */
+	private void setUpText(EditingDomain editingDomain) {
+		GridData gridData = new GridData(SWT.FILL, SWT.FILL, true, true);
+		gridData.heightHint = 100;
+	    textDescription.setLayoutData(gridData);
+	}
+
+	/**
+	 * Method to set up the label
+	 * @param label The label to be set up
+	 */
+	private void setUpLabel(Label label) {
+	    label.setLayoutData(createDefaultGridData());
+	}
+
+	/**
+	 * Method to set up the Icon
+	 */
+	private void setUpIcon() {
+		labelPropertyIcon.setLayoutData(new GridData());
+		Image dummyImage = emip.getOkImage();
+		labelPropertyIcon.setImage(dummyImage);
+	}
+	
+	@Override
+	public void setDataBinding(DataBindingContext dbCtx, EditingDomain editingDomain, EObject model) {
+		super.setDataBinding(dbCtx, editingDomain, model);
+		if (textDescription != null) {
+			@SuppressWarnings("unchecked")
+			IValueProperty<EObject, ?> descriptionProperty = EMFEditProperties.value(editingDomain, GeneralPackage.Literals.IDESCRIPTION__DESCRIPTION);
+			dbCtx.bindValue(WidgetProperties.text(SWT.Modify).observe(textDescription), descriptionProperty.observe(model));
+		}
+	}
+
+	@Override
+	public void updateState(boolean state) {
+		super.updateState(state);
+		
+		Image problemImage = emip.getProblemImageForStructuralFeatureInEobject(model, GeneralPackage.Literals.IDESCRIPTION__DESCRIPTION);
+		if (problemImage == null) {
```

Comment:
Is this really code which is individual to this snippet or should it be generic code in one of the super classes?

Reply:
Not required code in description snippet so removed it